src/boot/efi/chid.c

```diff
@@ -0,0 +1,385 @@
+        case 8:
+                status = hash_strings_sha1(
+                                &hash,
+                                (uint8_t *) &namespace,
+                                sizeof(namespace),
+                                7,
+                                info->manufacturer,
+                                L"&",
+                                info->product_name,
+                                L"&",
+                                info->baseboard_manufacturer,
+                                L"&",
+                                info->baseboard_product);
+                break;
+        case 9:
+                status = hash_strings_sha1(
+                                &hash,
+                                (uint8_t *) &namespace,
+                                sizeof(namespace),
+                                3,
+                                info->manufacturer,
+                                L"&",
+                                info->product_name);
+                break;
+        case 10:
+                status = hash_strings_sha1(
+                                &hash,
+                                (uint8_t *) &namespace,
+                                sizeof(namespace),
+                                7,
+                                info->manufacturer,
+                                L"&",
+                                info->family,
+                                L"&",
+                                info->baseboard_manufacturer,
+                                L"&",
+                                info->baseboard_product);
+                break;
+        case 11:
+                status = hash_strings_sha1(
+                                &hash,
+                                (uint8_t *) &namespace,
+                                sizeof(namespace),
+                                3,
+                                info->manufacturer,
+                                L"&",
+                                info->family);
+                break;
+        case 13:
+                status = hash_strings_sha1(
+                                &hash,
+                                (uint8_t *) &namespace,
+                                sizeof(namespace),
+                                5,
+                                info->manufacturer,
+                                L"&",
+                                info->baseboard_manufacturer,
+                                L"&",
+                                info->baseboard_product);
+                break;
+        case 14:
+                status = hash_strings_sha1(
+                                &hash, (uint8_t *) &namespace, sizeof(namespace), 1, info->manufacturer);
+                break;
+        default:
+                return EFI_SUCCESS; /* Just keep empty to prevent match. */
+        }
+        if (EFI_STATUS_IS_ERROR(status))
+                return status;
+
+        memcpy(chid, hash, sizeof(*chid));
```

Comment:
would be good having an assert_cc() here that validates that `sizeof(hash) >= sizeof(*chid)`